apps/x509.c

```diff
@@ -1097,3 +1105,93 @@ static int purpose_print(BIO *bio, X509 *cert, X509_PURPOSE *pt)
     }
     return 1;
 }
+
+static int parse_ext_names(char *names, const char **result)
+{
+    char *p, *q;
+    int cnt = 0, idx = 0, len = 0;
+
+    p = q = names;
+    len = strlen(names) + 1;
+
+    while (q - names <= len) {
```

Comment:
It also seems weird that an argument of "a," will return 1 whereas ",a" returns 2.